sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -232,28 +231,39 @@ private void performLogging(LogLevel logLevel, LogLevel environmentLogLevel, boo
      * Determines if the environment and logger support logging at the given log level.
      *
      * @param logLevel Logging level for the log message.
-     * @param environmentLoggingLevel Logging level the environment is set to support.
      * @return Flag indicating if the environment and logger are configured to support logging at the given log level.
      */
-    private boolean canLogAtLevel(LogLevel logLevel, LogLevel environmentLoggingLevel) {
+    public boolean canLogAtLevel(LogLevel logLevel) {
+        LogLevel rootLogLevel = loadSystemLogLevel();
+        return canLogAtLevel(logLevel, rootLogLevel);
+    }
+
+    private boolean canLogAtLevel(LogLevel logLevel, LogLevel allowedLogLevel) {
+
         // Attempting to log at a level not supported by the environment.
-        if (logLevel.toNumeric() < environmentLoggingLevel.toNumeric()) {
-            return false;
-        }
+        return logLevel.toNumeric() >= allowedLogLevel.toNumeric();
+    }
 
-        // Determine if the logger configuration supports logging at the level.
-        switch (logLevel) {
-            case VERBOSE:
-                return logger.isDebugEnabled();
-            case INFORMATIONAL:
-                return logger.isInfoEnabled();
-            case WARNING:
-                return logger.isWarnEnabled();
-            case ERROR:
-                return logger.isErrorEnabled();
-            default:
-                return false;
+    /**
+     * Checking the system log with the preference order of slf4j and environment variable.
+     *
+     * @return The log level.
+     */
+    private LogLevel loadSystemLogLevel() {
+        if (logger.isDebugEnabled()) {
```

Comment:
Should the binder still be checked? That way if an SLF4J binding exists but all logging levels are disabled we won't default into the environment configuration and instead wouldn't log anything as expected.

Reply:
Based on my testing, if there is some other logging binding to SLF4J, `logger.isDebugEnabled()` will reflect the log level set by user. So if log4j set log level to info, then the clientLogger `logger.canLogAtLevel(LogLevel.INFO)` returns true. 

Also, Binder check API is inside of `slf4j-simple` pkg which asks to introduce slf4j impl library to src code. This is not ideal as you comment above.